web/client/components/mapcontrols/searchbookmarkconfig/AddNewBookmark.jsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Button, Col, ControlLabel, FormControl as FC, FormGroup, Glyphicon, OverlayTrigger, Tooltip} from "react-bootstrap";
+import Message from "../../I18N/Message";
+import {inRange} from 'lodash';
+import localizedProps from '../../misc/enhancers/localizedProps';
+const FormControl = localizedProps('placeholder')(FC);
+
+const validate = (bookmark = {}) =>{
+    const {options = {}, title = ''} = bookmark;
+    const {south, north, west, east} = options;
+    const yValid = inRange(south, -90, 91) && inRange(north, -90, 91);
+    const xValid = inRange(west, -180, 181) && inRange(east, -180, 181);
+    const titleValid = !!title.trim().length;
+    return xValid && yValid && titleValid;
+};
+
+const AddNewBookmark = (props) => {
+    const {onPropertyChange, bookmark = {}, bbox: currentBBox} = props;
+    const {options: bboxAttributes = {}, title, layerVisibilityReload = false} = bookmark;
+
+    const onChange = (event) => {
+        const {value, name} = event.target;
+        const options = {...bookmark.options,  [name]: parseFloat(value)};
+        onPropertyChange("bookmark", {...bookmark, options});
+    };
+
+    const setTitle = (value) => {
+        onPropertyChange("bookmark", {...bookmark, title: value});
+    };
+
+    const applyCurrentBBox = () =>{
+        const [minX, minY, maxX, maxY] = currentBBox;
+        const bboxProperties = {
+            west: parseFloat(minX) || 0,
+            south: parseFloat(minY) || 0,
+            east: parseFloat(maxX) || 0,
+            north: parseFloat(maxY) || 0
+        };
+
+        const options = {...bookmark.options, ...bboxProperties};
+        onPropertyChange("bookmark", {...bookmark, options});
+    };
+
+    const validateFunc = (name) =>{
+        if (bboxAttributes[name]) {
+            if (['north', 'south'].includes(name)) {
+                return bboxAttributes[name] && !inRange(bboxAttributes[name], -90, 91) ? "error" : null;
+            }
+            return bboxAttributes[name] && !inRange(bboxAttributes[name], -180, 181) ? "error" : null;
+        }
+        return null;
+    };
+
+    const toggleLayerVisibility = () =>{
+        onPropertyChange("bookmark", {...bookmark, layerVisibilityReload: !layerVisibilityReload});
+    };
+
+    return (
```

Comment:
the coordinate part is similar to the one present in geometry details, maybe in the future we can unify them